### PR TITLE
AP_TemperatureSensor: fix TSYS01 warning on SITL

### PR DESCRIPTION
**Pull request by khancyr**
prevent "TSYS01 device is null!" warning on SITL !

### CODE REVIEW

File: libraries/AP_TemperatureSensor/TSYS01.cpp [hunk 1]
```diff
@@ -15,6 +15,9 @@ static const uint8_t TSYS01_CMD_READ_ADC    = 0x00;
 
 bool TSYS01::init()
 {
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    return true;
```
Comment (WickedShell) [inline libraries/AP_TemperatureSensor/TSYS01.cpp]: This should probably be false right? We didn't actually init the sensor.

Comment (khancyr) [reply libraries/AP_TemperatureSensor/TSYS01.cpp]: I think it should be true as on SITL, we are considering that the sensor is healthy and report a temperature.
@jaxxzer, on your call !

Comment (khancyr) [reply libraries/AP_TemperatureSensor/TSYS01.cpp]: ping @jaxxzer

Comment (jaxxzer) [reply libraries/AP_TemperatureSensor/TSYS01.cpp]: I think that we should make it false; then this is a non-functional change (besides eliminating the warnings). We don't have support in ::_read_adc() for sitl.

**Comment by khancyr** [comment]
Comment address, ready to merge

**Pull request merged**